src/SyntaxHighlighter/X86Assembly.cpp

```diff
@@ -216,9 +219,35 @@ const QRegularExpression kPlatformRegex{"^Platform:.*$"};
 X86Assembly::X86Assembly() : QSyntaxHighlighter{static_cast<QObject*>(nullptr)} {}
 
 void X86Assembly::highlightBlock(const QString& code) {
-  HighlightBlockAssembly(code, [this](int start, int count, const QTextCharFormat& format) {
-    setFormat(start, count, format);
-  });
+  const auto* const highlighting_metadata =
+      static_cast<const HighlightingMetadata*>(currentBlock().userData());
```

Comment:
Since we can't ensure that userdata is always of type `const HighlightingMetadata*` we should use `dynamic_cast` here.

Reply:
Yes, you are right!